### PR TITLE
add：connect nats userinfo option

### PR DESCRIPTION
**Pull request by CodingJzy**
* When the input type is nats, support username and password authentication

### CODE REVIEW

File: lib/input/reader/nats.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ import (
 // NATSConfig contains configuration fields for the NATS input type.
 type NATSConfig struct {
 	URLs          []string `json:"urls" yaml:"urls"`
+	UserName      string   `json:"user_name" yaml:"username"`
```
Comment (Jeffail) [inline lib/input/reader/nats.go:22]:
```suggestion
	Username      string   `json:"username" yaml:"username"`
```

File: lib/input/reader/nats.go [hunk 2]
```diff
@@ -87,7 +89,12 @@ func (n *NATS) ConnectWithContext(ctx context.Context) error {
 	var natsSub *nats.Subscription
 	var err error
 
-	if natsConn, err = nats.Connect(n.urls); err != nil {
+	opts := []nats.Option{}
+	if n.conf.UserName != "" || n.conf.Password != "" {
```
Comment (Jeffail) [inline lib/input/reader/nats.go:93]:
```suggestion
	if n.conf.Username != "" || n.conf.Password != "" {
```

**Review by Jeffail — CHANGES_REQUESTED** [review]
Hey @CodingJzy, looking good. Added a couple of comments just to make the field names more consistent. If we're adding auth fields to one nats component then I'd also want to make sure we add them to all others (the `nats` output and also the `nats_steam` input and output).

You can also delete `nats_test.go` as the integration tests are here: https://github.com/Jeffail/benthos/blob/master/lib/test/integration/nats_test.go

**Comment by CodingJzy** [comment]
Ok ! thanks for your guidance

**Comment by CodingJzy** [comment]
@Jeffail  hey :grin:  In fact. I found that nats and nats-streaming officially support url semantics. e.g. nats://root:pass@localhost:4222

**Comment by Jeffail** [comment]
> @Jeffail hey grin In fact. I found that nats and nats-streaming officially support url semantics. e.g. nats://root:pass@localhost:4222

Ahhh, in that case it's probably cleaner to document that syntax rather than add more fields, I've added examples with: https://github.com/Jeffail/benthos/commit/64ad8d480b9eb99e458ed543375b84c63c8211ea

Closing this PR for now but thanks for raising it and for the investigation.

**Pull request closed without merging**